### PR TITLE
SONARJAVA-2070 Support StringUtils methods

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java [hunk 1]
```diff
@@ -106,6 +115,32 @@ private boolean isObjectsRequireNonNullMethod(Symbol symbol) {
...
     }
 
+    @CheckForNull
+    private MethodBehavior createStringUtilMethodBehavior(Symbol.MethodSymbol symbol) {
+      MethodBehavior behavior;
+      switch (symbol.name()) {
+        case "isEmpty" :
+        case "isBlank" :
+        case "isNotEmpty" :
+        case "isNotBlank" :
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java]: MAJOR Reduce this switch case number of lines from 12 to at most 5, for example by extracting code into methods. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1151)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java [hunk 2]
```diff
@@ -106,6 +115,32 @@ private boolean isObjectsRequireNonNullMethod(Symbol symbol) {
...
+          behavior = new MethodBehavior(symbol);
+          MethodYield nullYield = new MethodYield(symbol.parameterTypes().size(), false);
+          nullYield.exception = false;
+          nullYield.parametersConstraints[0] = ObjectConstraint.nullConstraint();
+          nullYield.resultConstraint = symbol.name().contains("Not") ? BooleanConstraint.FALSE : BooleanConstraint.TRUE;
```
Comment (saberduck) [inline java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java]: I find having a condition on case values inside a switch statement confusing. Would be better to actually extract the method with parameter and call it with argument in right case branch

File: java-frontend/src/test/files/se/StringUtilsMethods.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    if(StringUtils.isNotBlank(param)) {
+      param.toString();
+    }
+  }
+
+  void otherMethod() {
+    String param = "";
+    StringUtils.chomp(param);
+  }
+}
```
Comment (saberduck) [inline java-frontend/src/test/files/se/StringUtilsMethods.java]: Missing EOL

**Pull request merged**